hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/OMKeyRequestFactory.java

```diff
@@ -66,43 +55,56 @@ public static OMKeyRequest createRequest(OMRequest omRequest,
     switch (cmdType) {
     case CreateDirectory:
       keyArgs = omRequest.getCreateDirectoryRequest().getKeyArgs();
-      omKeyRequest = OMDirectoryCreateRequest
-          .getInstance(keyArgs, omRequest, ozoneManager);
+      omKeyRequest = BucketLayoutAwareOMKeyRequestFactory.createRequest(
+          keyArgs.getVolumeName(), keyArgs.getBucketName(), omRequest,
+          ozoneManager);
       break;
     case CreateFile:
       keyArgs = omRequest.getCreateFileRequest().getKeyArgs();
-      omKeyRequest =
-          OMFileCreateRequest.getInstance(keyArgs, omRequest, ozoneManager);
+      omKeyRequest = BucketLayoutAwareOMKeyRequestFactory.createRequest(
+          keyArgs.getVolumeName(), keyArgs.getBucketName(), omRequest,
+          ozoneManager);
       break;
     case CreateKey:
       keyArgs = omRequest.getCreateKeyRequest().getKeyArgs();
-      omKeyRequest =
-          OMKeyCreateRequest.getInstance(keyArgs, omRequest, ozoneManager);
+      omKeyRequest = BucketLayoutAwareOMKeyRequestFactory.createRequest(
+          keyArgs.getVolumeName(), keyArgs.getBucketName(), omRequest,
+          ozoneManager);
       break;
     case AllocateBlock:
       keyArgs = omRequest.getAllocateBlockRequest().getKeyArgs();
-      omKeyRequest = OMAllocateBlockRequest
-          .getInstance(keyArgs, omRequest, ozoneManager);
+      omKeyRequest = BucketLayoutAwareOMKeyRequestFactory.createRequest(
+          keyArgs.getVolumeName(), keyArgs.getBucketName(), omRequest,
+          ozoneManager);
       break;
     case CommitKey:
       keyArgs = omRequest.getCommitKeyRequest().getKeyArgs();
-      omKeyRequest =
-          OMKeyCommitRequest.getInstance(keyArgs, omRequest, ozoneManager);
+      omKeyRequest = BucketLayoutAwareOMKeyRequestFactory.createRequest(
+          keyArgs.getVolumeName(), keyArgs.getBucketName(), omRequest,
+          ozoneManager);
       break;
     case DeleteKey:
       keyArgs = omRequest.getDeleteKeyRequest().getKeyArgs();
-      omKeyRequest =
-          OMKeyDeleteRequest.getInstance(keyArgs, omRequest, ozoneManager);
+      omKeyRequest = BucketLayoutAwareOMKeyRequestFactory.createRequest(
+          keyArgs.getVolumeName(), keyArgs.getBucketName(), omRequest,
+          ozoneManager);
       break;
     case DeleteKeys:
       return new OMKeysDeleteRequest(omRequest);
```

Comment:
@JyotinderSingh Please add this also to the factory class.

   ```
 case DeleteKeys:
      OzoneManagerProtocolProtos.DeleteKeyArgs deleteKeyArgs =
          omRequest.getDeleteKeysRequest().getDeleteKeys();
      omKeyRequest = BucketLayoutAwareOMKeyRequestFactory.createRequest(
          deleteKeyArgs.getVolumeName(), deleteKeyArgs.getBucketName(), omRequest,
          ozoneManager);
      break;
```

Reply:
Done